Special characters in filenames causing exceptions (Malformed input or input contains unmappable characters)

Got a [report](https://www.universalmediaserver.com/forum/viewtopic.php?t=16253) that filenames with special characters (like é, ñ, ü etc) cause errors and prevent anything showing when browsing folders with these files.

```
ERROR 2024-11-08 21:35:24.006 [Media Scanner] net.pms.store.MediaScanner Unhandled exception during media scan: Malformed input or input contains unmappable characters: /muziekarchief/MUSIC/Bach/Bach complete org (MP3)/Clavier-��bung III (Bach 2000 v08CD06)
TRACE 2024-11-08 21:35:24.007 [Media Scanner] net.pms.store.MediaScanner 
java.nio.file.InvalidPathException: Malformed input or input contains unmappable characters: /muziekarchief/MUSIC/Bach/Bach complete org (MP3)/Clavier-��bung III (Bach 2000 v08CD06)
	at java.base/sun.nio.fs.UnixPath.encode(Unknown Source)
	at java.base/sun.nio.fs.UnixPath.<init>(Unknown Source)
	at java.base/sun.nio.fs.UnixFileSystem.getPath(Unknown Source)
	at java.base/java.nio.file.Path.of(Unknown Source)
	at java.base/java.nio.file.Paths.get(Unknown Source)
	at net.pms.store.container.VirtualFolder.lambda$getFilesListForDirectories$0(VirtualFolder.java:154)
	at java.base/java.io.File.listFiles(Unknown Source)
	at net.pms.store.container.VirtualFolder.getFilesListForDirectories(VirtualFolder.java:153)
	at net.pms.store.container.VirtualFolder.discoverChildren(VirtualFolder.java:215)
	at net.pms.store.MediaScanner.scan(MediaScanner.java:141)
	at net.pms.store.MediaScanner.scan(MediaScanner.java:147)
	at net.pms.store.MediaScanner.scan(MediaScanner.java:147)
	at net.pms.store.MediaScanner.startScan(MediaScanner.java:98)
	at net.pms.store.MediaScanner.lambda$startMediaScan$0(MediaScanner.java:198)
	at java.base/java.lang.Thread.run(Unknown Source)
ERROR 2024-11-08 21:35:47.648 [webgui-server-40] net.pms.network.webguiserver.servlets.PlayerApiServlet Exception in PlayerApiServlet: Malformed input or input contains unmappable characters: /muziekarchief/MUSIC/Bach/Bach complete org (MP3)/Clavier-��bung III (Bach 2000 v08CD06)
TRACE 2024-11-08 21:35:47.649 [webgui-server-40] net.pms.network.webguiserver.servlets.PlayerApiServlet {}
java.nio.file.InvalidPathException: Malformed input or input contains unmappable characters: /muziekarchief/MUSIC/Bach/Bach complete org (MP3)/Clavier-��bung III (Bach 2000 v08CD06)
	at java.base/sun.nio.fs.UnixPath.encode(Unknown Source)
	at java.base/sun.nio.fs.UnixPath.<init>(Unknown Source)
	at java.base/sun.nio.fs.UnixFileSystem.getPath(Unknown Source)
	at java.base/java.nio.file.Path.of(Unknown Source)
	at java.base/java.nio.file.Paths.get(Unknown Source)
	at net.pms.store.container.VirtualFolder.lambda$getFilesListForDirectories$0(VirtualFolder.java:154)
	at java.base/java.io.File.listFiles(Unknown Source)
	at net.pms.store.container.VirtualFolder.getFilesListForDirectories(VirtualFolder.java:153)
	at net.pms.store.container.VirtualFolder.discoverChildren(VirtualFolder.java:215)
	at net.pms.store.StoreContainer.discover(StoreContainer.java:696)
	at net.pms.store.MediaStore.getResources(MediaStore.java:664)
	at net.pms.network.webguiserver.servlets.PlayerApiServlet.getBrowsePage(PlayerApiServlet.java:339)
	at net.pms.network.webguiserver.servlets.PlayerApiServlet.doPost(PlayerApiServlet.java:153)
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:547)
	at net.pms.network.webguiserver.GuiHttpServlet.service(GuiHttpServlet.java:61)
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:614)
	at org.eclipse.jetty.ee10.servlet.ServletHolder.handle(ServletHolder.java:736)
	at org.eclipse.jetty.ee10.servlet.ServletHandler$ChainEnd.doFilter(ServletHandler.java:1614)
	at org.eclipse.jetty.ee10.servlet.ServletHandler$MappedServlet.handle(ServletHandler.java:1547)
	at org.eclipse.jetty.ee10.servlet.ServletChannel.dispatch(ServletChannel.java:824)
	at org.eclipse.jetty.ee10.servlet.ServletChannel.handle(ServletChannel.java:436)
	at org.eclipse.jetty.ee10.servlet.ServletHandler.handle(ServletHandler.java:464)
	at org.eclipse.jetty.server.handler.ContextHandler.handle(ContextHandler.java:1060)
	at org.eclipse.jetty.server.Server.handle(Server.java:181)
	at org.eclipse.jetty.server.internal.HttpChannelState$HandlerInvoker.run(HttpChannelState.java:661)
	at org.eclipse.jetty.server.internal.HttpConnection.onFillable(HttpConnection.java:406)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:322)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:99)
	at org.eclipse.jetty.io.SelectableChannelEndPoint$1.run(SelectableChannelEndPoint.java:53)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.runTask(AdaptiveExecutionStrategy.java:478)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.consumeTask(AdaptiveExecutionStrategy.java:441)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.tryProduce(AdaptiveExecutionStrategy.java:293)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.run(AdaptiveExecutionStrategy.java:201)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:311)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:979)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.doRunJob(QueuedThreadPool.java:1209)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1164)
	at java.base/java.lang.Thread.run(Unknown Source)
ERROR 2024-11-08 21:36:19.042 [webgui-server-38] net.pms.network.webguiserver.servlets.PlayerApiServlet Exception in PlayerApiServlet: Malformed input or input contains unmappable characters: /muziekarchief/MUSIC/Christmas/Andre Rieu/Merry Christmas/09 - Minuit Chr��tien.mp3
TRACE 2024-11-08 21:36:19.045 [webgui-server-38] net.pms.network.webguiserver.servlets.PlayerApiServlet {}
java.nio.file.InvalidPathException: Malformed input or input contains unmappable characters: /muziekarchief/MUSIC/Christmas/Andre Rieu/Merry Christmas/09 - Minuit Chr��tien.mp3
	at java.base/sun.nio.fs.UnixPath.encode(Unknown Source)
	at java.base/sun.nio.fs.UnixPath.<init>(Unknown Source)
	at java.base/sun.nio.fs.UnixFileSystem.getPath(Unknown Source)
	at java.base/java.nio.file.Path.of(Unknown Source)
	at java.base/java.nio.file.Paths.get(Unknown Source)
	at net.pms.store.container.VirtualFolder.lambda$getFilesListForDirectories$0(VirtualFolder.java:154)
	at java.base/java.io.File.listFiles(Unknown Source)
	at net.pms.store.container.VirtualFolder.getFilesListForDirectories(VirtualFolder.java:153)
	at net.pms.store.container.VirtualFolder.discoverChildren(VirtualFolder.java:215)
	at net.pms.store.StoreContainer.discover(StoreContainer.java:696)
	at net.pms.store.MediaStore.getResources(MediaStore.java:664)
	at net.pms.network.webguiserver.servlets.PlayerApiServlet.getBrowsePage(PlayerApiServlet.java:339)
	at net.pms.network.webguiserver.servlets.PlayerApiServlet.doPost(PlayerApiServlet.java:153)
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:547)
	at net.pms.network.webguiserver.GuiHttpServlet.service(GuiHttpServlet.java:61)
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:614)
	at org.eclipse.jetty.ee10.servlet.ServletHolder.handle(ServletHolder.java:736)
	at org.eclipse.jetty.ee10.servlet.ServletHandler$ChainEnd.doFilter(ServletHandler.java:1614)
	at org.eclipse.jetty.ee10.servlet.ServletHandler$MappedServlet.handle(ServletHandler.java:1547)
	at org.eclipse.jetty.ee10.servlet.ServletChannel.dispatch(ServletChannel.java:824)
	at org.eclipse.jetty.ee10.servlet.ServletChannel.handle(ServletChannel.java:436)
	at org.eclipse.jetty.ee10.servlet.ServletHandler.handle(ServletHandler.java:464)
	at org.eclipse.jetty.server.handler.ContextHandler.handle(ContextHandler.java:1060)
	at org.eclipse.jetty.server.Server.handle(Server.java:181)
	at org.eclipse.jetty.server.internal.HttpChannelState$HandlerInvoker.run(HttpChannelState.java:661)
	at org.eclipse.jetty.server.internal.HttpConnection.onFillable(HttpConnection.java:406)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:322)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:99)
	at org.eclipse.jetty.io.SelectableChannelEndPoint$1.run(SelectableChannelEndPoint.java:53)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.runTask(AdaptiveExecutionStrategy.java:478)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.consumeTask(AdaptiveExecutionStrategy.java:441)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.tryProduce(AdaptiveExecutionStrategy.java:293)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.run(AdaptiveExecutionStrategy.java:201)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:311)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:979)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.doRunJob(QueuedThreadPool.java:1209)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1164)
	at java.base/java.lang.Thread.run(Unknown Source)
```

I use 14.11.0. Here is a log extract exhibiting the problem:

`Exception in thread "main" java.lang.ExceptionInInitializerError
	at net.pms.store.MediaScanner.init(MediaScanner.java:79)
	at net.pms.PMS.init(PMS.java:269)
	at net.pms.PMS.createInstance(PMS.java:1009)
	at net.pms.PMS.main(PMS.java:960)
Caused by: java.nio.file.InvalidPathException: Malformed input or input contains unmappable characters: /home/me/Vid?os`

Where the folder is /home/me/Vidéos

--- a/src/main/java/net/pms/configuration/GuiConfiguration.java
+++ b/src/main/java/net/pms/configuration/GuiConfiguration.java
@@ -28,6 +28,7 @@ import java.io.IOException;
 import java.io.Serializable;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
+import java.nio.file.InvalidPathException;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import org.apache.commons.lang3.StringUtils;
@@ -41,6 +42,8 @@ public class GuiConfiguration implements Serializable {
 
 	private static final Logger LOGGER = LoggerFactory.getLogger(GuiConfiguration.class);
 	private static final Gson GSON = new GsonBuilder().setPrettyPrinting().create();
+	private static final String ERROR_CONF_FILE = "Error in gui configuration file : ";
+
 	private String graphicsDevice;
 	private Rectangle screenBounds;
 	private Insets screenInsets;
@@ -159,15 +162,22 @@ public class GuiConfiguration implements Serializable {
 	 */
 	public static GuiConfiguration getConfiguration() {
 		GuiConfiguration conf = null;
-		Path path = Paths.get(UmsConfiguration.getProfileDirectory()).resolve("GUI.conf");
+		Path path = null;
 		try {
+			path = Paths.get(UmsConfiguration.getProfileDirectory()).resolve("GUI.conf");
 			if (Files.exists(path)) {
 				LOGGER.info("Getting gui configuration file : " + path);
 				String json = Files.readString(path, StandardCharsets.UTF_8);
 				conf = GSON.fromJson(json, GuiConfiguration.class);
 			}
-		} catch (IOException | JsonSyntaxException ex) {
-			LOGGER.info("Error in gui configuration file : " + ex.getMessage());
+		} catch (IOException ex) {
+			LOGGER.info(ERROR_CONF_FILE + ex.getMessage());
+			LOGGER.debug(null, ex);
+		} catch (InvalidPathException ex) {
+			LOGGER.info(ERROR_CONF_FILE + "invalid path");
+			LOGGER.debug(null, ex);
+		} catch (JsonSyntaxException ex) {
+			LOGGER.info(ERROR_CONF_FILE + "json is malformed");
 			LOGGER.debug(null, ex);
 		}
 		if (conf == null) {

--- a/src/main/java/net/pms/configuration/old/OldConfigurationImporter.java
+++ b/src/main/java/net/pms/configuration/old/OldConfigurationImporter.java
@@ -18,6 +18,7 @@ package net.pms.configuration.old;
 
 import java.io.File;
 import java.nio.file.Files;
+import java.nio.file.InvalidPathException;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.ArrayList;
@@ -135,32 +136,36 @@ public class OldConfigurationImporter {
 			if (KEY_FOLDERS.equals(key)) {
 				LOGGER.info("Checking shared folder: \"{}\"", folder);
 			}
-
-			Path path = Paths.get(folder);
-			if (Files.exists(path)) {
-				if (!Files.isDirectory(path)) {
-					if (KEY_FOLDERS.equals(key)) {
-						LOGGER.warn(
-							"The \"{}\" is not a folder! Please remove it from your shared folders " +
-							"list on the \"Shared Content\" tab or in the configuration file.",
-							folder
-						);
-					} else {
-						LOGGER.debug("The \"{}\" is not a folder - check the configuration for key \"{}\"", folder, key);
+			try {
+				Path path = Paths.get(folder);
+				if (Files.exists(path)) {
+					if (!Files.isDirectory(path)) {
+						if (KEY_FOLDERS.equals(key)) {
+							LOGGER.warn(
+								"The \"{}\" is not a folder! Please remove it from your shared folders " +
+								"list on the \"Shared Content\" tab or in the configuration file.",
+								folder
+							);
+						} else {
+							LOGGER.debug("The \"{}\" is not a folder - check the configuration for key \"{}\"", folder, key);
+						}
 					}
+				} else if (KEY_FOLDERS.equals(key)) {
+					LOGGER.warn(
+						"\"{}\" does not exist. Please remove it from your shared folders " +
+						"list on the \"Shared Content\" tab or in the configuration file.",
+						folder
+					);
+				} else {
+					LOGGER.debug("\"{}\" does not exist - check the configuration for key \"{}\"", folder, key);
 				}
-			} else if (KEY_FOLDERS.equals(key)) {
-				LOGGER.warn(
-					"\"{}\" does not exist. Please remove it from your shared folders " +
-					"list on the \"Shared Content\" tab or in the configuration file.",
-					folder
-				);
-			} else {
-				LOGGER.debug("\"{}\" does not exist - check the configuration for key \"{}\"", folder, key);
-			}
 
-			// add the path even if there are problems so that the user can update the shared folders as required.
-			folders.add(path);
+				// add the path even if there are problems so that the user can update the shared folders as required.
+				folders.add(path);
+			} catch (InvalidPathException ex) {
+				LOGGER.info("Invalid path found: {}", folder);
+				LOGGER.debug(null, ex);
+			}
 		}
 
 		return folders;

--- a/src/main/java/net/pms/configuration/sharedcontent/SharedContentConfiguration.java
+++ b/src/main/java/net/pms/configuration/sharedcontent/SharedContentConfiguration.java
@@ -24,8 +24,10 @@ import com.google.gson.JsonSyntaxException;
 import com.sun.jna.Platform;
 import java.io.File;
 import java.io.IOException;
+import java.nio.charset.MalformedInputException;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
+import java.nio.file.InvalidPathException;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.ArrayList;
@@ -50,6 +52,7 @@ public class SharedContentConfiguration {
 		.create();
 	private static final List<SharedContentListener> LISTENERS = new ArrayList<>();
 	private static final SharedContentArray SHARED_CONTENT_ARRAY = new SharedContentArray();
+	private static final String ERROR_CONF_FILE = "Error in shared content configuration file : ";
 
 	// Automatic reloading
 	private static boolean isWritingConfiguration = false;
@@ -238,8 +241,8 @@ public class SharedContentConfiguration {
 	 *    - /user
 	 */
 	private static synchronized void readAllConfigurations() {
-		Path sharedConfFilePath = Paths.get(CONFIGURATION.getSharedConfPath());
 		try {
+			Path sharedConfFilePath = Paths.get(CONFIGURATION.getSharedConfPath());
 			if (Files.exists(sharedConfFilePath)) {
 				LOGGER.info("Getting shared content from configuration file : " + sharedConfFilePath);
 				String json = Files.readString(sharedConfFilePath, StandardCharsets.UTF_8);
@@ -256,24 +259,38 @@ public class SharedContentConfiguration {
 				}
 				updateSharedContent(oldConfig, true);
 			}
-		} catch (IOException | JsonSyntaxException ex) {
-			LOGGER.info("Error in shared content configuration file : " + ex.getMessage());
+		} catch (MalformedInputException ex) {
+			LOGGER.info(ERROR_CONF_FILE + "file is not UTF-8 charset");
+			LOGGER.debug(null, ex);
+			updateSharedContent(new SharedContentArray(), false);
+		} catch (JsonSyntaxException ex) {
+			LOGGER.info(ERROR_CONF_FILE + "json is malformed");
+			LOGGER.debug(null, ex);
+			updateSharedContent(new SharedContentArray(), false);
+		} catch (InvalidPathException | IOException ex) {
+			LOGGER.info(ERROR_CONF_FILE + ex.getMessage());
 			LOGGER.debug(null, ex);
 			updateSharedContent(new SharedContentArray(), false);
 		}
 	}
 
 	private static synchronized SharedContentArray readConfiguration() {
-		Path sharedConfFilePath = Paths.get(CONFIGURATION.getSharedConfPath());
-		LOGGER.debug("Reading shared content configuration file: " + sharedConfFilePath);
 		try {
+			Path sharedConfFilePath = Paths.get(CONFIGURATION.getSharedConfPath());
+			LOGGER.debug("Reading shared content configuration file: " + sharedConfFilePath);
 			if (Files.exists(sharedConfFilePath)) {
 				String json = Files.readString(sharedConfFilePath, StandardCharsets.UTF_8);
 				return GSON.fromJson(json, SharedContentArray.class);
 			}
 			LOGGER.trace("Shared content configuration file missing: " + sharedConfFilePath);
-		} catch (IOException | JsonSyntaxException ex) {
-			LOGGER.info("Error in shared content configuration file : " + ex.getMessage());
+		} catch (MalformedInputException ex) {
+			LOGGER.info(ERROR_CONF_FILE + "file is not UTF-8 charset");
+			LOGGER.debug(null, ex);
+		} catch (JsonSyntaxException ex) {
+			LOGGER.info(ERROR_CONF_FILE + "json is malformed");
+			LOGGER.debug(null, ex);
+		} catch (InvalidPathException | IOException ex) {
+			LOGGER.info(ERROR_CONF_FILE + ex.getMessage());
 			LOGGER.debug(null, ex);
 		}
 		return new SharedContentArray();
@@ -300,8 +317,8 @@ public class SharedContentConfiguration {
 			Path sharedConfFilePath = Paths.get(CONFIGURATION.getSharedConfPath());
 			LOGGER.debug("Writing shared content configuration file: " + sharedConfFilePath);
 			Files.writeString(sharedConfFilePath, GSON.toJson(SHARED_CONTENT_ARRAY), StandardCharsets.UTF_8);
-		} catch (IOException e) {
-			LOGGER.debug("An error occurred while writing the shared content configuration file: {}", e);
+		} catch (InvalidPathException | IOException ex) {
+			LOGGER.debug("An error occurred while writing the shared content configuration file: {}", ex);
 		}
 		isWritingConfiguration = false;
 	}

--- a/src/main/java/net/pms/platform/windows/WindowsUtils.java
+++ b/src/main/java/net/pms/platform/windows/WindowsUtils.java
@@ -499,7 +499,7 @@ public class WindowsUtils extends PlatformUtils {
 			}
 			vlcPath = Paths.get(Advapi32Util.registryGetStringValue(WinReg.HKEY_LOCAL_MACHINE, key, ""));
 			vlcVersion = new Version(Advapi32Util.registryGetStringValue(WinReg.HKEY_LOCAL_MACHINE, key, "Version"));
-		} catch (Win32Exception e) {
+		} catch (InvalidPathException | Win32Exception e) {
 			LOGGER.debug("Could not get VLC information from Windows registry: {}", e.getMessage());
 			LOGGER.trace("", e);
 		}

--- a/src/main/java/net/pms/store/container/VirtualFolder.java
+++ b/src/main/java/net/pms/store/container/VirtualFolder.java
@@ -17,9 +17,6 @@
 package net.pms.store.container;
 
 import java.io.File;
-import java.nio.file.Files;
-import java.nio.file.Path;
-import java.nio.file.Paths;
 import java.util.*;
 import java.util.Map.Entry;
 import net.pms.configuration.sharedcontent.VirtualFolderContent;
@@ -150,24 +147,29 @@ public class VirtualFolder extends StoreContainer {
 			}
 
 			if (directory.canRead()) {
-				File[] listFiles = directory.listFiles((File parentDirectory, String file) -> {
-					Path path = Paths.get(parentDirectory + File.separator + file);
+				File[] listFiles = directory.listFiles((File file) -> {
 					// Reject any non readable
-					if (!Files.isReadable(path)) {
+					if (!file.canRead()) {
+						LOGGER.trace("Ignoring '{}' because it is unreadable", file);
 						return false;
 					}
+					String filename = file.getName();
 					// Accept any directory
-					if (Files.isDirectory(path)) {
+					if (file.isDirectory()) {
 						// Skip if ignored
-						if (!ignoredDirectoryNames.isEmpty() && ignoredDirectoryNames.contains(file)) {
-							LOGGER.debug("Ignoring {} because it is in the ignored directories list", file);
-							return false;
+						boolean ignored = (!ignoredDirectoryNames.isEmpty() && ignoredDirectoryNames.contains(filename));
+						if (ignored) {
+							LOGGER.debug("Ignoring '{}' because it is in the ignored directories list", file);
 						}
-						return true;
+						return !ignored;
 					}
 
 					// We want to find only media files
-					return SystemFilesHelper.isPotentialMediaFile(file);
+					boolean isMediaFile = SystemFilesHelper.isPotentialMediaFile(filename);
+					if (!isMediaFile) {
+						LOGGER.trace("Ignoring '{}' because it is not a media file", file);
+					}
+					return isMediaFile;
 				});
 
 				if (listFiles == null) {

--- a/src/main/java/net/pms/util/FileWatcher.java
+++ b/src/main/java/net/pms/util/FileWatcher.java
@@ -25,6 +25,7 @@ import java.nio.file.FileSystems;
 import java.nio.file.FileVisitOption;
 import java.nio.file.FileVisitResult;
 import java.nio.file.Files;
+import java.nio.file.InvalidPathException;
 import java.nio.file.Path;
 import java.nio.file.PathMatcher;
 import java.nio.file.Paths;
@@ -91,7 +92,7 @@ public class FileWatcher {
 			} else {
 				add(w, dir);
 			}
-		} catch (NullPointerException e) {
+		} catch (InvalidPathException | NullPointerException e) {
 			LOGGER.info("Not watching invalid path {} for changes", w.getFileSpec());
 		}
 	}

